gusto/transport_equation.py

```diff
@@ -290,55 +362,43 @@ def __init__(self, state, V, ibp="twice", equation_form="advective", supg_params
                          solver_params=solver_params,
                          outflow=outflow)
 
+        if is_dg(V):
+            raise ValueError("You are trying to apply SUPG on a discontinuous space")
         # if using SUPG we either integrate by parts twice, or not at all
-        if ibp == "once":
+        if ibp == IntegrateByParts.ONCE:
             raise ValueError("if using SUPG we don't integrate by parts once")
-        if ibp is None and not self.is_cg:
+        if ibp == IntegrateByParts.NEVER and not is_cg(V):
             raise ValueError("are you very sure you don't need surface terms?")
 
         # set default SUPG parameters
-        dt = state.timestepping.dt
-        supg_params = supg_params.copy() if supg_params else {}
-        supg_params.setdefault('ax', dt/sqrt(15.))
-        supg_params.setdefault('ay', dt/sqrt(15.))
-        supg_params.setdefault('az', dt/sqrt(15.))
-        # default assumes a continuous space
-        supg_params.setdefault('dg_direction', None)
-
-        # find out if we need to do DG upwinding in any direction and set
-        # self.dS accordingly
-        if supg_params["dg_direction"] is None:
-            # space is assumed to be continuous and we don't need
-            # any interior surface integrals
-            self.dS = None
-        elif supg_params["dg_direction"] == "horizontal":
-            # if space is discontinuous in the horizontal direction, we
-            # need to include surface integrals on the vertical faces
-            self.dS = dS_v
-        elif supg_params["dg_direction"] == "vertical":
-            # if space is discontinuous in the vertical direction, we
-            # need to include surface integrals on the horizontal faces
-            self.dS = dS_h
+        if supg_params is None:
+            supg_params = SUPGOptions()
+
+        dim = state.mesh.topological_dimension()
+        if supg_params.tau is not None:
+            # if tau is provided, check that is has the right size
+            tau = supg_params.tau
+            assert tau.ufl_shape == (dim, dim)
```

Comment:
Might be helpful to also return an error message. This will just raise an `AssertionError`. Something like:
```python
assert tau.ufl_shape == (dim, dim), "Provided tau has incorrect shape!"
```
Also, what if someone provides a `tau` which is not a UFL expression? It won't have a `ufl_shape` and trigger an `AttributeError`. One option would be to raise an error if it isn't, or simply cast via `as_ufl(supg_params.tau)`.

Reply:
Yes, good point - thanks!